documentation/projects/proposals/trust_and_safety/detecting_sensitive_textual_content/20230506-implementation_plan_frontend_blurring_sensitive.md

```diff
@@ -0,0 +1,264 @@
+value of `is_sensitive` for each media item.
+
+### Step 4: Blurring in search results
+
+The approaches to concealment of sensitive content in the search result page is
+specific to the media type. For image results, the image itself is blurred; for
+audio results, the title is blurred.
+
+The cell components for search results, i.e. image cell and two layouts (box and
+row) of the audio track will be updated to automatically blur the right parts of
+the UI, based on the store state of the
+[`blur_sensitive` toggle](#what-about-the-blur-toggle) and the `is_sensitive`
+value for their respective media item.
+
+We are currently going with the CSS blur approach.
+
+### Step 5: Blurring in single result pages
+
+The single result page will be updated to render a safety wall when the media
+item is sensitive. The safety wall will be a full page overlay that will conceal
+the entire content of the page.
+
+It will have a button that will allow the user to proceed to see the content or
+to go to a default page.
+
+There is a difference in the behaviour of this page based on whether the user
+has arrived to it from a search result page or directly. In the former case, the
+preference for the [`blur_sensitive` toggle](#what-about-the-blur-toggle) will
+be carried over to the single result page (as they will be in the same session),
+and if the user has opted to see sensitive content unblurred, the wall will not
+be shown to them.
+
+Care must be taken to not show the content at all unless we have the information
+about the sensitivity of the media item. A flash of clearly visible sensitive
+content before the wall appears will be jarring and will defeat the purpose of
+the wall.
+
+## Design
+
+<!-- Note any design requirements for this plan. -->
+
+The necessary designs for this plan were proposed and extensively iterated in
+[WordPress/openverse#791](https://github.com/WordPress/openverse/issues/791).
+
+The finalised designs can be found in Figma document for
+[Safe content browsing flow mockups](https://www.figma.com/file/bDTfGzBit8rGRPktOR1cu4/Safe-content-browsing-flow?type=design&node-id=0-1).
+
+As of writing the document, the design work has been completed.
+
+## Parallelizable streams
+
+<!-- What, if any, work within this plan can be parallelized? -->
+
+The development of the required components can be parallelized. The last step,
+which will be bringing all the code together (to put the sensitive content
+behind the wall when the user has not opted, or opted out, of seeing sensitive
+material) is the one that will be blocked till all the pieces have come
+together.
+
+## Blockers
+
+<!-- What hard blockers exist which might prevent further work on this project? -->
+
+This project will need the `include_sensitive_results` parameter to no longer be
+unstable so that it can be used by the frontend. The API also needs to supply
+the `sensitivity` field which will tell the frontend whether to blur or not.
```

Comment:
The feature could be developed with the parameter and field still marked as `unstable__`, right? We'd just have to coordinate updating the usage before enabling the frontend flag for the feature in production, I suppose.

Reply:
I feel a little uncomfortable using an `unstable__` flag in the frontend, which is definitely a production environment. Since we're maintaining both the API and the frontend, we can coordinate breaking changes but even so, ideally, we'd want the flag to be stable before the GA of the feature.

Reply:
Updated the blockers in fc24b7d.

Reply:
> we'd want the flag to be stable before the GA of the feature

Certainly. The reason to develop frontend initially with the unstable feature is to make sure that the API changes are complete for our own purposes and work as expected before stabilising (and therefore committing) to a specific behaviour that ends up not even covering our _own_ requirements.